include/libcyphal/transport/can/transport.hpp

```diff
@@ -442,9 +480,110 @@ class TransportImpl final : public ICanTransport, private TransportDelegate
+        const auto         is_anonymous      = local_node_id > CANARD_NODE_ID_MAX;
+        const std::size_t  total_active_subs = total_msg_subscriptions_ + (is_anonymous ? 0 : total_svc_subscriptions_);
+
+        // There is no memory allocation here yet - just empty span.
+        //
+        cetl::span<Filter>                  filters_span{};
+        libcyphal::detail::VarArray<Filter> filters{total_active_subs, &memory()};
+        if (total_active_subs > 0)
+        {
+            // Now we know that we have at least one active subscription,
+            // so we need preallocate temp memory for total number of active subscriptions.
+            //
+            filters.reserve(total_active_subs);
+            if (filters.capacity() < total_active_subs)
+            {
+                // This is out of memory situation. We will just leave this run,
+                // but `should_reconfigure_filters_` will stay engaged, so we will try again on next run.
+                return;
+            }
+
+            // `subs_count` counting is just for the sake of debug verification.
+            std::size_t            subs_count = 0;
+            CanardTreeNode** const subs_trees = canard_instance().rx_subscriptions;
+
+            if (total_msg_subscriptions_ > 0)
+            {
+                const auto msg_visitor = [&filters](RxSubscription& rx_subscription) {
+                    const auto msb_flt = ::canardMakeFilterForSubject(rx_subscription.port_id);
+                    filters.emplace_back(Filter{msb_flt.extended_can_id, msb_flt.extended_mask});
+                };
+                subs_count += RxSubscriptionTree::visitCounting(subs_trees[CanardTransferKindMessage], msg_visitor);
+            }
+
+            // No need to build service filters if we don't have a local node ID.
+            //
+            if ((total_svc_subscriptions_ > 0) && !is_anonymous)
+            {
+                const auto svc_visitor = [&filters, local_node_id](RxSubscription& rx_subscription) {
+                    const auto flt = ::canardMakeFilterForService(rx_subscription.port_id, local_node_id);
+                    filters.emplace_back(Filter{flt.extended_can_id, flt.extended_mask});
+                };
+                subs_count += RxSubscriptionTree::visitCounting(subs_trees[CanardTransferKindRequest], svc_visitor);
+                subs_count += RxSubscriptionTree::visitCounting(subs_trees[CanardTransferKindResponse], svc_visitor);
+            }
+
+            (void) subs_count;
+            CETL_DEBUG_ASSERT(subs_count == total_active_subs, "");
+
+            filters_span = {filters.data(), filters.size()};
+        }
+
+        // Let know each media interface about the new filters (tracking the fact of possible media error).
+        //
+        bool was_error = false;
+        for (const Media& media : media_array_)
+        {
+            const cetl::optional<MediaError> maybe_error = media.interface.setFilters(filters_span);
```

Comment:
I don't understand what do we need the `filters_span` for? Can't we just construct it as a temporary ad-hoc instead of introducing a new named entity? It should be implicitly constructible from the array.

```suggestion
            const cetl::optional<MediaError> maybe_error = media.interface.setFilters(filters);
```

Reply:
No we can't:
```
/Users/sergei/Develop/git/OpenCyphal-Garage/libcyphal/include/libcyphal/transport/can/transport.hpp:567:87: error: no viable conversion from 'libcyphal::detail::VarArray<Filter>' (aka 'VariableLengthArray<libcyphal::transport::can::Filter, polymorphic_allocator<libcyphal::transport::can::Filter>>') to 'Filters' (aka 'span<const libcyphal::transport::can::Filter, 18446744073709551615UL>')
            const cetl::optional<MediaError> maybe_error = media.interface.setFilters(filters);
                                                                                      ^~~~~~~
```

, but we you right about not needed `filters_span`: we can do like this:
```
const cetl::optional<MediaError> maybe_error = media.interface.setFilters({filters.data(), filters.size()});
```